services/sources/publisher_backlist_service.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+import json
+import urllib.parse
+from typing import Optional, Dict
+
+from logger import create_log
+from mappings.UofM import UofMMapping
+from .source_service import SourceService
+
+
+logger = create_log(__name__)
+
+BASE_URL = "https://api.airtable.com/v0/appBoLf4lMofecGPU/Publisher%20Backlists%20%26%20Collections%20%F0%9F%93%96?view=UofMichigan%20Press"
+
+class PublisherBacklistService(SourceService):
+    def __init__(self):
+        self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
+
+    def get_records(
+        self,
+        full_import: bool=False, 
+        start_timestamp: datetime=None,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ) -> list[UofMMapping]:
+        
+        array_json_records = self.get_records_json(full_import, start_timestamp, offset, limit)
+
+        array_index = 0
+        while array_index != len(array_json_records):
```

Comment:
could we use a for loop here instead to remove the array_index increment line?

Reply:
Sure